scripts/output/export/xlsx_IIASA.R

```diff
@@ -9,234 +9,221 @@ require(data.table)
-
-message("# ", length(variables_not_in_template), " variables not in IIASA template are deleted, see ", LOGFILE, ".")
-write(paste0("\n\n#--- ", length(variables_not_in_template), " variables not in IIASAtemplate ", iiasatemplate,
-             " are deleted ---#"), file = LOGFILE, append = TRUE)
-write(paste0("  - ", paste(variables_not_in_template, collapse = "\n  - ")), file = LOGFILE, append = TRUE)
-
-message("# ", length(temporarydelete), " variables are in the list to be temporarily deleted, ",
-        length(unique(data$variable[data$variable %in% temporarydelete])), " were deleted, see ", LOGFILE, ".")
-write(paste0("\n\n#--- ", length(variables_not_in_template), " variables are in temporarydelete. ",
-      length(unique(data$variable[data$variable %in% temporarydelete])), " were deleted ---#"),
-      file = LOGFILE, append = TRUE)
-write(paste0("  - ", paste(unique(data$variable[data$variable %in% temporarydelete]), collapse = "\n  - ")), file = LOGFILE, append = TRUE)
-
-data <- data %>%
-  mutate(value = ifelse(!is.finite(value) | is.na(value), 0, value)) %>%
-  mutate(scenario=gsub("NA", "", scenario)) %>%
-  filter(variable %in% templatevariables) %>%
-  filter(! variable %in% temporarydelete)
-
-# some basic checks
-scenarios <- unique(data$variable)
-for (i in 1:length(scenarios)) {
-  stopifnot(length(filter(data, variable %in% scenarios[[1]])) == length(filter(data, variable %in% scenarios[[i]])))
-}
+  # check whether all scenarios have same number of variables
+  scenarios <- unique(mifdata$variable)
+  for (i in 1:length(scenarios)) {
+    stopifnot(length(filter(mifdata, variable %in% scenarios[[1]])) == length(filter(mifdata, variable %in% scenarios[[i]])))
+  }
+
+  file.remove(OUTPUT_mif)
+  write.mif(mifdata, OUTPUT_mif)
+
+  # perform summation checks
+  checkSummationData <- checkSummations(OUTPUT_mif, template = mapping, summationsFile = "AR6",
+                                        logFile = logFile, logAppend = TRUE,
+                                        dataDumpFile = file.path(outputFolder, "checkSummations.csv"))
 
-file.remove(OUTPUT_mif)
-write.mif(data, OUTPUT_mif)
+  mifdata %>%
+    mutate(value = ifelse(!is.finite(value) | paste(value) == "", 0, value)) %>%
+    pivot_wider(names_from="period", values_from="value") -> writetoexcel
 
-# filter(scenario == "d_delfrag") %>%
+  writexl::write_xlsx(list("data" = writetoexcel), OUTPUT_xlsx)
 
-data %>%
-  mutate(value = ifelse(!is.finite(value) | paste(value) == "", 0, value)) %>%
-  pivot_wider(names_from="period", values_from="value") -> writetoexcel
+  message("\n### mif files: ", sum(existingfiles), " used, ", sum(! existingfiles), " not found, see above.")
+  message("\n### Output files written:\n- ", OUTPUT_mif, "\n- ", OUTPUT_xlsx)
 
-writexl::write_xlsx(list("data" = writetoexcel), OUTPUT_xlsx)
+}, message = function(x) {
+  cat(x$message, file = logFile, append = TRUE)
```

Comment:
Will this also still print to console or only to logFile?

Reply:
it prints also to console. I don't know exactly, why, to be fair, but what you suggested produced two prints to the console.

Reply:
Okay good. I'd guess there's a chain of handlers, and after your custom handler it will just continue with the default handler which prints to console.